system_tests/bigquery.py

```diff
@@ -90,7 +92,15 @@ def test_update_dataset(self):
         after = [grant for grant in dataset.access_grants
                  if grant.entity_id != 'projectWriters']
         dataset.access_grants = after
-        dataset.update()
+
+        # We should try and keep `delay` and tries as low as possible to
+        # reduce test running time. tries=3 and delay=30 worked consistenly
+        # when updating a dataset.
```

Comment:
Oh gotcha. No idea there. I'd guess the rate-limit is just based on total number of requests, but we could try to find a service owner to ask. We also hit the API back-to-back quickly in Py2 and Py3.

ASIDE: Shouldn't it be 429 (`TooManyRequests`)? Or is this one of the APIs that overuses the wrong status codes?